### PR TITLE
feat(*): add preliminary download-build-pull logic

### PR DESCRIPTION
**Pull request by arschles**
addresses part of https://github.com/deis/builder/issues/7

boot.go builds a `dockerbuilder` cli, which:
1. creates a temporary directory (`tmp/` for purposes of this example)
2. uses [`mc`](http://github.com/minio/mc) to download a tarball from `$TAR_URL`
3. untars the tarball from previous
4. runs `docker build -t $IMG_NAME .` on the untarred codebase
5. runs `docker push $IMG_NAME` on the newly built image
6. removes `tmp/` and everything in it

on any error, prints an error message, cleans up `tmp/` and all contents and exits 1

cc/ @smothiki @technosophos

### CODE REVIEW

**Comment by arschles** [comment]
also needs to be integrated with https://github.com/deis/builder/pull/8

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.5
+FROM alpine:3.1
```
Comment (technosophos) [inline Dockerfile]: `alpine:3.2`?

**Comment by technosophos** [comment]
Just two comments:
1. I think we've been trying to move all the Alpine stuff to 3.2.
2. How come we're checking in the mc binary instead of fetching it on build?

**Comment by arschles** [comment]
@technosophos moved to `alpine:3.2`, but don't know of any place from which to download the `mc` binary. I've checked it into a few repositories just to get things working for now and I have a TODO to put it somewhere and download it. I'll create issues on the applicable repositories to download it instead of commit it

**Comment by krancour** [comment]
Commit style...

**Comment by arschles** [comment]
I really messed up here. I'll amend and squash a few. thanks

**Comment by arschles** [comment]
@krancour amended

File: manifests/deis-dockerbuilder-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    app: deis-dockerbuilder
+  template:
+    metadata:
+      labels:
+        # Important: these labels need to match the selector above
+        app: deis-dockerbuilder
+    spec:
+      containers:
+        - name: deis-dockerbuilder
+        image: arschles/dockerbuilder:0.0.1-20151124102330
```
Comment (smothiki) [inline manifests/deis-dockerbuilder-rc.yaml:24]: This should be POD not RC

File: manifests/deis-dockerbuilder-service.yaml [hunk 3]
```diff
@@ -13,4 +15,4 @@ spec:
       protocol: TCP
   selector:
     # Match the selector in the RC
-    app: deis-example
+    app: deis-dockerbuilder
```
Comment (smothiki) [inline manifests/deis-dockerbuilder-service.yaml:18]: This project doesn't require any service file

File: manifests/deis-dockerbuilder-rc.yaml (hunk 2 above)
Comment (smothiki) [inline manifests/deis-dockerbuilder-rc.yaml:24]: POD template should be something similar to slugbuilder POD template

**Comment by smothiki** [comment]
@arschles Do you still want to continue working on this. I have no luck with Go but it's still do able with the help of tar archive and input streams and our own implementation of docker socket client. 
Apparently writing in python makes developing effort less.

**Comment by arschles** [comment]
@smothiki as much as I love Go, you've already built, tested and merged the python version. I'm ok with leaving as-is and closing this.

**Pull request closed without merging**